### DISCUSSION TITLE
How to use environment variables?

### DISCUSSION THREAD

**Discussion by muffl0n** [stepci/stepci#25] [Q&A] [answered]
In your example, you set an env-var:
```
env:
  host: example.com
```
I'd expect, that I can override/set this value (no matter if it's set in the workflow.yml or not). But this does not seem to work:
```
/ # cat tests/workflow.yaml
version: "1.0"
name: Status Check
env:
  host: example.com
tests:
  example:
    steps:
      - name: GET request
        url: https://{{env.host}}
        method: GET
        check:
          status: /^20/
/ # export host=httpbin.org
/ # echo $host
httpbin.org
/ # stepci run /tests/workflow.yaml

 GET request  PASSED  in 0.614s

Request

 GET  https://example.com/  200 OK

Checks

Status
✔ 200

✔ example passed in 0.614s

✔ Status Check (/tests/workflow.yaml) passed in 0.649s
```

Am I doing it wrong?

**Comment by mishushakov** [comment]
You are not doing anything wrong! We haven't implemented env variable passing yet and are looking for contributors, who could help us out!

Please see this issue: https://github.com/stepci/stepci/issues/9
You can also subscribe to the issue to get a notification when we have this resolved

We're sorry for the inconvenience 😢

**Comment by mishushakov** [comment, marked as answer]
We added support for environment variables in the current version of stepci, thanks to a generous contribution from @npostulart
See pull request https://github.com/stepci/stepci/pull/33

1. Install the latest version of stepci available

    ```
    npm install -g stepci@latest
    ```

2. Run your workflow with custom environment variables

   ```
   stepci run workflow.yml -e env=$VAR
   ```

3. Additionally, see our [CLI Reference](https://github.com/stepci/stepci/blob/main/docs/cli-reference.md)

Hopefully this answers your question!

**Reply by muffl0n** [reply]
Awesome! Thank you very much!

**Comment by sebastianwis** [comment]
Hi Sven @muffl0n 
Thank you for your contribution here! 😊
It would be amazing to welcome you to our Discord if you haven't joined it yet:
https://discord.com/invite/KqJJzJ3BTu
This is where we keep users up to date with updates, features and bug resolutions.
